internal/replay.go

```diff
@@ -256,8 +258,12 @@ const averageReplayDuration = 20 * time.Minute
 
 func (r *ReplayCmd) verifySLOs(ctx context.Context, replays []ReplayConfig) error {
 	sloNames := make([]string, 0, len(replays))
-	for i := range replays {
-		sloNames = append(sloNames, replays[i].SLO)
+	for _, r := range replays {
+		sloNames = append(sloNames, r.SLO)
+		if r.SourceSLO != nil {
+			// Add source SLOs to the list of SLOs to check for existence and permissions.
+			sloNames = append(sloNames, r.SourceSLO.Slo)
```

Comment:
Further, this function checks if data source of this SLO supports Replay. This will disallow to Replay SLO A from SLO B if B's data source doesn't support Replay even though we are not using B's data source at all.

We could relax this verification and not check source's data source for replay availability.

Reply:
As we talked:
* we agreed this doesn't check for `sourceSLO` availability so it won't be disallowed
* we agreed that in fact we don't need to check for data source availability for target SLO but since it would bring on us more work to do, (because the endpoint for availability also performs checks for replay concurrency limits) we're ok to go with performing checks for datasource even if replay with sourceSLO does not need that check.